### ISSUE TITLE
`# noqa` has odd behaviour in blocks

### ISSUE THREAD

**Issue by peterchoo** [ansible/ansible-lint#588]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible 2.5.5
ansible-lint 4.1.0
```

- ansible installation method: pip
- ansible-lint installation method: pip

# Desired Behaviour

`# noqa` works as expected in blocks, specifically for rule 305.

# Actual Behaviour (Bug report only)

As per #516, `# noqa` doesn't work as expected in blocks.

Playbook:

```yaml
---
  
- name: Test Playbook
  hosts: all
  tasks:
    - name: Shell Command  # noqa 305
      shell: echo "test"
      
    - block:
        - name: Shell Command 2  # noqa 305
          shell: echo "test 2"
```

Ansible Lint output:

```
$ ansible-lint playbook.yml 
[301] Commands should not change things if nothing needs doing
playbook.yml:6
Task/Handler: Shell Command

[305] Use shell only when shell functionality is required
playbook.yml:10
```

Rule 301 was surprisingly not picked up in the block, and 305 wasn't skipped.

**Comment by xiaopeng163** [comment]
this works for me

```
---

- name: Test Playbook
  hosts: all
  tasks:
    - name: Shell Command  # noqa 305 301
      shell: echo "test"

    - block:
        - name: Shell Command 2  # noqa 305
          shell: echo "test 2"
```
```
$ ansible --version
ansible 2.8.7
$ ansible-lint --version
ansible-lint 4.2.0

**Comment by ssbarnea** [comment]
@webknjaz safe to close, works fine in last release

**Comment by mnasiadka** [comment]
I'm getting the same in 5.4.0 - although it's not numeric rule, but package-latest...
